src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -35,7 +36,30 @@ public class RestoreCommand
 
         public Guid ParentId { get; }
 
-        public const string ProjectRestoreInformation = "ProjectRestoreInformation";
+        // names for ProjectRestoreInformation and intervals
+        private const string ProjectRestoreInformation = "ProjectRestoreInformation";
+        private const string ErrorCodes = "ErrorCodes";
+        private const string ErrorMessages = "ErrorMessages";
+        private const string WarningCodes = "WarningCodes";
+        private const string WarningMessages = "WarningMessages";
+        private const string RestoreSuccess = "RestoreSuccess";
+
+        // names for child events for ProjectRestoreInformation
+        private const string GenerateRestoreGraph = "GenerateRestoreGraph";
+        private const string GenerateAssetsFile = "GenerateAssetsFile";
+        private const string ValidateRestoreGraphs = "ValidateRestoreGraphs";
+        private const string CreateRestoreResult = "CreateRestoreResult";
+        private const string RestoreNoOpInformation = "RestoreNoOpInformation";
+
+        // names for intervals in RestoreNoOpInformation
+        private const string CacheFileEvaluateDuration = "CacheFileEvaluateDuration";
+        private const string AssetsVerificationDuration = "AssetVerificationDuration";
```

Comment:
add msbuild here...it's kind of confusing otherwise.

Reply:
what is confusing. can you elaborate?